app/controllers/users_controller.rb

```diff
@@ -22,9 +27,49 @@ def create
     end
   end
 
+  def update
+    if @user.update(user_params)
+      flash[:success] = t "profile_updated"
+      redirect_to @user
+    else
+      flash[:danger] = t "update_fail"
+      render :edit
+    end
+  end
+
+  def destroy
+    if @user&.destroy
+      flash[:success] = t "user_deleted"
+    else
+      flash[:danger] = t "delete_fail"
+    end
+    redirect_to users_url
+  end
+
   private
 
+  def set_user
+    @user = User.find_by(id: params[:id])
+    return if @user
+
+    flash[:danger] = t "user_not_found"
+    redirect_to root_url
+  end
+
   def user_params
-    params.require(:user).permit :name, :email, :password
+    prs = [:name, :email, :password, :password_confirmation]
+    params.require(:user).permit prs
```

Comment:
chịu khó quá

tại User model tạo 1 biến hằng `REGISTER_ATTRS = %i(name email password password_confirmation)`
Sau đó, ở đây chỉ cần 
```suggestion
    params.require(:user).permit User::REGISTER_ATTRS
```